DataManagementSystem/DB/FileCatalogComponents/FileManager/FileManagerPs.py

```diff
@@ -775,51 +775,51 @@ def _setFileParameter(self, fileID, paramName, paramValue, connection=False):
   # _getFileReplicas related methods
   #
 
-  def _getFileReplicas(self, fileIDs, fields_input=['PFN'], allStatus=False, connection=False):
+  def _getFileReplicas(self, fileIDs, fields_input=None, allStatus=False, connection=False):
     """ Get replicas for the given list of files specified by their fileIDs
         :param fileIDs : list of file ids
-        :param fields_input : metadata of the Replicas we are interested in
+        :param fields_input : metadata of the Replicas we are interested in (default to PFN)
         :param allStatus : if True, all the Replica statuses will be considered,
                            otherwise, only the db.visibleReplicaStatus
 
         :returns S_OK with a dict { fileID : { SE name : dict of metadata } }
     """
 
-    connection = self._getConnection(connection)
+    if fields_input is None:
+      fields_input = ['PFN']
 
     fields = list(fields_input)
 
+    # always add Status in the list of required fields
     if 'Status' not in fields:
       fields.append('Status')
 
-    replicas = {}
+    # We initialize the dictionnary with empty dict
+    # as default value, because this is what we want for
+    # non existing replicas
+    replicas = {fileID: {} for fileID in fileIDs}
```

Comment:
what if you use `collections.defaultdict(dict)` ?

Reply:
Then files without replicas won't be included in the result.

Reply:
yes, exactly